src/shared/modules/connections/connectionsDuck.ts

```diff
@@ -630,7 +638,66 @@ export const connectionLostEpic = (action$: any, store: any) =>
     .filter(() => inWebEnv(store.getState()) && isConnected(store.getState()))
     .throttleTime(5000)
     .do(() => store.dispatch(updateConnectionState(PENDING_STATE)))
-    .mergeMap(() => {
+    .mergeMap((action: any) => {
+      if (action.error.code === 'Neo.ClientError.Security.TokenExpired') {
+        const SSOProviders = getActiveConnectionData(store.getState())
+          ?.SSOProviders
+        if (SSOProviders) {
+          return Rx.Observable.fromPromise(
+            (async function() {
+              //TODO can throw?
```

Comment:
yeah, if the `handleRefreshingToken` throws, we should bail out of the reconnection attempts, and go straight to the connection lost state. TBD is whether we should specifically report this back to the user somehow, possible with custom error text?